### DISCUSSION TITLE
Changed IP/SSH Port, GUI no longer works

### DISCUSSION THREAD

**Discussion by giganovatech** [StackStorm/st2#5454] [General]
I changed the IP and SSH port of the machine where stackstorm resides last night. I can still login to the GUI but it is now blank and I get error messages in the bottom lefthand corner that say things like "Unable to retrieve history. See details in developer tools console stackstorm." 

Also, if I try to do something like this: st2 action list
I get an error like this: ERROR: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))

How do I fix this?

**Comment by arm4b** [comment]
Check if the backends behind the StackStorm are available and working OK:
* MongoDB
* RabbitMQ
* Redis

most likely one of those is down.

**Comment by minsis** [comment]
You may also need to change the IP addresses in `/opt/stackstorm/static/webui/st2web/config.js` if your Web UI is running on  a different node than the rest of st2

**Comment by giganovatech** [comment]
@armab I ran the following commands: systemctl status mongod, systemctl status nginx, systemctl status redis, systemctl status rabbitmq-server and all of the services had the same status: Active: active (running). I also tried restarting all of the services. No change in the behavior of the web ui.

@minsis the web ui and st2 run on the same box. I went ahead and checked relevant config files for st2, mongodb, nginx, redis and rabbitmq-server and didn't see anything related to the previous ip, not that I couldn't have missed something.

Any other ideas or should I just reinstall st2? I haven't had it running for long so I haven't built much and it should be easy to get back to the state I have it in now.

**Reply by minsis** [reply]
Only thing I can think of is maybe there's a misconfig in `/etc/st2/st2.conf` but you said you already checked that. It simply sounds like something somewhere is configured to still point to the old IP address and/or listen on the old IP address. Have you tried restarting st2, nginx, redis, rabbitmq and mongo?

You can run something like `netstat -utlpn | grep 91` to see if the st2 web services are running on the correct port and listening on the correct IP. `netstat -utlpn | grep mongo` might show you what IP address that mongo, redis, and rabbitmq are listening on.

On the local machine you can try and run `curl -v http://localhost:9100` (or https), `curl -v http://localhost:9101` and `curl -v http://localhost:9102` to validate that api, auth, and stream are listening properly. If you get a json response then the services are listening. Also, maybe replace `localhost` with the new and old IP address of the machine. From another machine in the same subnet try to `curl  -v http://<new-ip>:9100`.

Also check `/var/log/st2/st2api.log` to see if there are any errors there.

Hopefully, one of these things can help you pin point some issues.

**Reply by minsis** [reply]
Also `st2ctl status` will give tell you if *all* the st2 services are up and running.
